### PR TITLE
Feat: lesson10

### PR DESCRIPTION
**Pull request by yuka830**
# [Lesson10](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#10)
## [CodeSandbox](https://codesandbox.io/s/lesson10-gn6i7?file=/main.js)
Last commit [75b7525](https://github.com/yuka830/js-lessons/pull/12/commits/75b75257f890f05989a4dc1c10c58f3745e6de70)

Adding try-catch-finally to lesson9

[Question]
I had no idea what I should write in `finally`. I just wrote "done", is it ok?

### CODE REVIEW

**Comment by ghost** [comment]
Originally, I think "loaded" should be written in "finally" since it is always deleted whether the communication succeeds or fails, but in this case, there were no instructions in the problem, so I approved it.

**Review by ghost — APPROVED** [review]

**Comment by yuka830** [comment]
@443n0511 
Thanks for your comment. I agree with your idea. Even if it fails, there is nothing wrong for `loaded` being executed. So I've changed :)
[70c3dfd](https://github.com/yuka830/js-lessons/pull/12/commits/70c3dfdd7f8e13f14c05f3064d063a2dc5b68d82)

File: lesson10/main.js
Comment (ghost) [inline lesson10/main.js]: Thank you for fixing this. I've already "Approve" it once, and I'll leave it to you to fix it or not, but this process alone will not display anything on the screen when an error occurs.
Here is the code I imagine.
 [Please take a look if you like.](https://codesandbox.io/s/lesson10-forked-8nbq8?file=/main.js).

Since the purpose of this assignment was to write "at the end", I wrote the exact same process as yuka-san and got approved during the teacher's review, so please take the sandbox as an idea😊

Comment (yuka830) [reply lesson10/main.js]: @443n0511 
Thank you for your advice! I thought that idea was nice, it's like a user friendly! I would like to adopt that.
My code became nice with your advice, much appreciated! :) :)
[bccf0fc](https://github.com/yuka830/js-lessons/pull/12/commits/bccf0fc28ea89ddbe4996389d6c3f39c96e62506)
[CodeSandbox](https://codesandbox.io/s/lesson10-gn6i7?file=/main.js)

File: lesson10/main.js
Comment (matsuwork) [inline lesson10/main.js]: IMO

I've been told by Mr.Morita the following

Do not execute "createNewList" in the try.

This is because you can’t judge
whether the error is from "fetchedData" or "createNewList".

Even if an error occurs from "createNewList", 
"データの取得ができませんでした。" will be displayed.

Comment (yuka830) [reply lesson10/main.js]: @matsuwork
Thank you for your advice.I did't realize that but that's a good point!
I searched how to solve and I got a hint. Here’s what I got, I'm not sure it's correct though.

- Implemented return in the try block
- moved `createNewList` from try block to global and executed with then method

[d217c13 ](https://github.com/yuka830/js-lessons/pull/12/commits/d217c1379d661eeb4196f1977704d15c36bf8008)
[CodeSandbox](https://codesandbox.io/s/lesson10-gn6i7?file=/main.js)

Comment (matsuwork) [reply lesson10/main.js]: The idea is correct.
Async await is recommended,
so I think it would be better to use that.

Comment (yuka830) [reply lesson10/main.js]: @matsuwork
Thank you for the reply. I rewrote `createNewList` part from then method to async/await!
Is this way correct?
[4664130](https://github.com/yuka830/js-lessons/pull/12/commits/46641304373e21b99f09adef43bad8a946620735)
[CodeSandbox](https://codesandbox.io/s/lesson10-gn6i7?file=/main.js)

File: lesson10/main.js
Comment (matsuwork) [inline lesson10/main.js]: Good. I approve.

nits:
I would prefer a different name than "result".
For example, "init".

**Review by matsuwork — APPROVED** [review]

File: lesson10/main.js
Comment (yuka830) [reply lesson10/main.js]: @matsuwork
I fixed name, thank you for teaching me and your time :)
[75b7525](https://github.com/yuka830/js-lessons/pull/12/commits/75b75257f890f05989a4dc1c10c58f3745e6de70)
[CodeSandbox](https://codesandbox.io/s/lesson10-gn6i7?file=/main.js)

**Pull request merged**
